### PR TITLE
Makefile: fix generating coverage when specifiying TESTPKGS

### PR DESCRIPTION
**Pull request by christarazi**
See commit msg for details. This should help surface coverage information when developing on individual packages, and such.

### CODE REVIEW

**Comment by christarazi** [comment]
test-me-please

**Comment by joestringer** [comment]
Do you think that this will cause the coveralls coverage to stabilize now? It's reporting a 4.6% drop in coverage but it can be pretty temperamental regardless of changes like this.

**Review by joestringer — CHANGES_REQUESTED** [review]
Assuming this works as intended and doesn't adversely impact the Coveralls integration in CI, I think the only other piece is that presumably the `COVERPKG` variable is no longer needed and can be removed from the git tree.

Alternatively when running locally you *can* specify `COVERPKG=...` to choose the package to apply coverage for, although while this works it's not a great workflow so if this PR solves the problem more generally I'd rather get rid of it.

**Comment by christarazi** [comment]
> Do you think that this will cause the coveralls coverage to stabilize now? It's reporting a 4.6% drop in coverage but it can be pretty temperamental regardless of changes like this.

I didn't realize that this change would affect this, but it seems like it would fix the weird fluctuations we see in coverage. 

> Assuming this works as intended and doesn't adversely impact the Coveralls integration in CI, I think the only other piece is that presumably the `COVERPKG` variable is no longer needed and can be removed from the git tree.

Removed them. 

> Alternatively when running locally you _can_ specify `COVERPKG=...` to choose the package to apply coverage for, although while this works it's not a great workflow so if this PR solves the problem more generally I'd rather get rid of it.

Decided to remove them because as an individual you can generate coverage for an individual package by specifying it in `TESTPKGS` with this change.

**Comment by christarazi** [comment]
test-me-please

**Comment by joestringer** [comment]
Looking at the [Coveralls Link](https://coveralls.io/builds/30548102) I wonder if we're basically changing whether we see coverage for only the individual unit tests vs. seeing coverage across all packages given the set of tests we run overall. Subtle difference but for instance the tests in `daemon/` will likely touch a whole bunch of other packages which these changes likely now exclude.

**Comment by christarazi** [comment]
> Looking at the [Coveralls Link](https://coveralls.io/builds/30548102) I wonder if we're basically changing whether we see coverage for only the individual unit tests vs. seeing coverage across all packages given the set of tests we run overall. Subtle difference but for instance the tests in `daemon/` will likely touch a whole bunch of other packages which these changes likely now exclude.

After spending some time digging, I think you're correct. 

So what we actually want is the following:

 1) Providing `TESTPKGS` will set `COVERPKG` to the current package being tested.
 2) Running the entire suite (`TESTPKGS` is empty), will set `COVERPKG` to `./...`

For item 1, the intent of this PR was to fix this. However, I fixed it in a more aggressive manner than I anticipated. 

For item 2, this is the current behavior today, so no need to do anything. 

Thoughts @joestringer?

**Comment by christarazi** [comment]
test-me-please

File: Makefile [hunk 1]
```diff
@@ -161,7 +161,8 @@ endif
 	# hence will trigger an error of too many arguments. As a workaround, we
 	# have to process these packages in different subshells.
 	for pkg in $(patsubst %,github.com/cilium/cilium/%,$(TESTPKGS)); do \
-		$(GO_TEST) $(TEST_UNITTEST_LDFLAGS) $$pkg $(GOTEST_BASE) $(GOTEST_COVER_OPTS) || exit 1; \
+		$(GO_TEST) $(TEST_UNITTEST_LDFLAGS) $$pkg $(GOTEST_BASE) $(GOTEST_COVER_OPTS) || \
+		exit 1; \
```
Comment (joestringer) [inline Makefile]: nit, is this just a stylistic change? Elsewhere in the Makefile we use a slightly different style to try to highlight the conditional:

```suggestion
		$(GO_TEST) $(TEST_UNITTEST_LDFLAGS) $$pkg $(GOTEST_BASE) $(GOTEST_COVER_OPTS) \
		|| exit 1; \
```

**Review by joestringer — CHANGES_REQUESTED** [review]
Minor style nit, otherwise looks good. Good to see a filesystem iteration getting removed here too, that should speed up builds.

File: Makefile (hunk 1 above)
Comment (christarazi) [reply Makefile]: Addressed

**Comment by christarazi** [comment]
test-me-please

**Comment by christarazi** [comment]
Coveralls is back to "normal". (I am still puzzled by the Coveralls percentage variance. :confused:)

**Review by joestringer — APPROVED** [review]

**Comment by joestringer** [comment]
Not sure how to trigger docs / mod gh actions but there are no changes in those areas so I think this is good to go.

**Pull request merged**